Add lead support engineer role (#22)
OctopusDeploy/People: Support.md

- I am actively introspective and take time to work on my personal and professional development.
- I happily take responsibility for my own mistakes and I learn from the mistakes of others.

## Lead Support Engineer

The Lead Support Engineer role builds upon the foundation of the Support Engineer and Senior Support Engineer roles.

> I am a technical leader, a person that is recognised both within the Support team and as part of the wider Octopus community for the breadth and depth of your technical ability.

### I am trusted

- I have mastered my tools and the core building blocks used by our team.

> - I have a a deep technical understanding of more than one area of Octopus, and can confidently use this knowledge to assist customers and fellow support team members in all areas of Octopus.

Reviewer: ```suggestion
- I have a deep technical understanding of more than one area of Octopus, and I can confidently use this knowledge to assist customers and fellow support team members in all areas of Octopus.
```

- I know when it makes sense to push back against a customer, and when it is appropriate to not create an issue

> - I know how to mentor a customer to the best path, and understand that what they are asking for may not be what they need

Reviewer: I tripped over the original statement when I read it, this is an attempt to make it a bit easier to parse (`that what` is correct but it still confused me):
```suggestion
- I know how to mentor a customer to the best path, and I understand that the customer's stated issue and the actual issue might not be the same thing
```

> 

Reviewer: Thinking about the progression from Support Engineer to Senior Support Engineer and on to Lead Support Engineer, I think it might be good to include something like:
```suggestion
- I am trusted across Octopus Deploy and have established good working relationships across the organization, and I co-ordinate our response to complicated support issues by pulling in the right people at the right time.
```

### I am a mentor of mentors

> - I know when to assist with a ticket, and when to push to encourage others to learn more.

Reviewer: I like this. There's an important distinction between `doing something wrong` and `not doing it the way I would have done it` which I think this point gets at, but I wonder if it's worth calling out. For instance:
- I encourage others to find creative solutions to problems and accept that my way isn't the only way.

Author: Excellent call Lee.

- I help others in my team to become mentors.
- I am always on the lookout for opportunities to teach and show others alternate, cleaner ways to delight our customers in an ego-less manner. 
- I am working to improve our processes, making it easier for everyone in our team to grow and learn.

> 

Reviewer 2: I assume this is similar to engineering leads vs managers.  The leads wouldn't do performance reviews.  The manager still would.

Should anything be mentioned a lead organizes a team of support engineers to solve a large support related problem?  Such as working on a way to easily spin up and down test targets or spin up and down test OD instances to test out customer configurations?

Author: I've added these two lines to cover those points without being too specific

- I actively research and implement technical solutions to improve our response times
- I confidently, accurately, and quickly replicate complex deployment scenarios to assist with issue replication and resolution

Reviewer 2: I like the updates!

### Everything is a learning oppurtunity

> 

Reviewer: This might be covered in the other points, but I think it's worth calling out for a lead position:
- I share the lessons I learn so the entire team and Octopus can benefit

Author: That's part of the Senior Role:
- I can effectively coach people and pass on my knowledge.
- I am a confident and proficient teacher of my craft.

But I'm going to add that specific line again anyway :)
